vlasovsolver/cpu_acc_transform.cpp

```diff
@@ -98,53 +109,329 @@ Eigen::Transform<Real,3,Eigen::Affine> compute_acceleration_transformation(
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " EJE 1 " << spatial_cell->parameters[CellParams::EYJE] << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " EJE 2 " << spatial_cell->parameters[CellParams::EZJE] << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " B 0 " << Bx << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " B 1 " << By << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " B 2 " << Bz << endl;
+
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBXdy " << dBXdy << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBXdz " << dBXdz << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBYdx " << dBYdx << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBYdz " << dBYdz << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBZdx " << dBZdx << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBZdy " << dBZdy << endl;
+	}*/
+   }  
+
+    // compute total transformation
+   Transform<Real,3,Affine> total_transform(Matrix<Real, 4, 4>::Identity());
 
-   unsigned int bulk_velocity_substeps; // in this many substeps we iterate forward bulk velocity when the complete transformation is computed (0.1 deg per substep).
-   bulk_velocity_substeps = fabs(dt) / fabs(gyro_period*(0.1/360.0));
-   if (bulk_velocity_substeps < 1) bulk_velocity_substeps=1;
+   // in this many substeps we iterate forward when the complete transformation is computed (0.1 deg per substep).
+   unsigned int transformation_substeps; 
+   transformation_substeps = fabs(dt) / fabs(gyro_period*(0.1/360.0));
+   //
+   if (smallparticle) {
+      unsigned int transformation_substeps_2; 
+      transformation_substeps_2 = fabs(dt / plasma_period*(0.1/360.0));
+      transformation_substeps = transformation_substeps_2 > transformation_substeps ? transformation_substeps_2 : transformation_substeps;
 
-   const Real substeps_radians = -(2.0*M_PI*dt/gyro_period)/bulk_velocity_substeps; // how many radians each substep is.
-   const Real substeps_dt=dt/bulk_velocity_substeps; /*!< how many s each substep is*/
+    /*ofstream substepFile;
+      substepFile.open ("substep_test.txt", ios::app);
+      if (dt > 0.001*fabs(gyro_period)) {
+      substepFile << " transformation_substeps = " << transformation_substeps << endl;
+      substepFile << " dt: " << dt << endl;
+      substepFile << " substeps_radians: " << substeps_radians << endl;
+      substepFile << " gyro and plasma periods: " << fabs(gyro_period) << "\t" << plasma_period << endl;
+      substepFile << " CellID: " << spatial_cell->parameters[CellParams::CELLID] << endl;
+      substepFile << " N: " << floor(dt/fabs(gyro_period)) << endl;
+      substepFile << endl;
+      }
+    */
```

Comment:
Is this still needed?

Reply:
again, debug stuff to be removed before merging to dev.